polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -217,6 +217,34 @@ $loading-panel-height: 53px;
     }
   }
 
+  &.TableRow-child {
+    // stylelint-disable -- Polaris component custom properties
+    --pc-index-table-checkbox-width: var(--p-space-500);
+    --pc-index-table-checkbox-offset-left: var(--p-space-300);
+    --pc-index-table-checkbox-border-width: var(--p-border-width-0165);
+    --pc-index-table-checkbox-child-offset: calc(
+      var(--pc-index-table-checkbox-offset-left) +
+        var(--pc-index-table-checkbox-width)
+    );
+    --pc-index-table-shifted-checkbox: 32;
+
+    .TableCell-first {
+      left: var(--pc-index-table-checkbox-child-offset);
+      z-index: var(--pc-index-table-shifted-checkbox);
+      @media #{$p-breakpoints-sm-down} {
+        left: calc(var(--pc-index-table-checkbox-child-offset));
+      }
+    }
+    .TableCell-first + .TableCell > span {
```

Comment:
If we add `left` to `.TableCell-first + .TableCell` it will result in this table cell runs over the next table cell